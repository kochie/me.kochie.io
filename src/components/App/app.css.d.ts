export const container: string;
export const app: string;
