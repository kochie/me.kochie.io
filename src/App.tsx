import * as React from "react";

import Sidebar from "./components/Sidebar";

import Main from "./components/Main";

import "./styles/App.scss";

export default class App extends React.Component {
  render() {
    return (
      <div className="container">
        <Sidebar />

        <Main />
      </div>
    );
  }
}
