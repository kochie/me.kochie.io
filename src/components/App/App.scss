@import "../../styles/constants";

body {
  margin: 0;
}

.container {
  display: grid;
  width: 100%;
  height: 100%;
  grid-template-columns: 33vw 1fr;
  grid-template-rows: auto;
  grid-template-areas: "sidebar main";
  @media only screen and (max-width: $mobile-width) and (max-height: $mobile-height) and (min-resolution: $mobile-resolution),
    only screen and (max-width: $mobile-height) and (max-height: $mobile-width) and (min-resolution: $mobile-resolution) {
    height: 100%;
    min-height: 100vh;
    grid-template-columns: auto;
    grid-template-rows: 100vh 1fr;
    grid-template-areas: "sidebar" "main";
  }

  // @media screen and (min-width: 1500px){
  //   grid-template-columns: 1fr 3fr;
  // }
}

.app {
  position: relative;
}
