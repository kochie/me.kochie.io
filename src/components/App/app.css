body {
  margin: 0;
}

.container {
  display: grid;
  width: 100%;
  height: 100%;
  grid-template-columns: 33vw 1fr;
  grid-template-rows: auto;
  grid-template-areas: "sidebar main";
}

@media (--tablet-size) {
  .container {
    height: 100%;
    min-height: 100vh;
    grid-template-columns: auto;
    grid-template-rows: 100vh 1fr;
    grid-template-areas: "sidebar" "main";
  }
}

@media (--mobile-size) {
  .container {
    height: 100%;
    min-height: 100vh;
    grid-template-columns: auto;
    grid-template-rows: 100vh 1fr;
    grid-template-areas: "sidebar" "main";
  }
}

.app {
  position: relative;
  overflow: hidden;
}
