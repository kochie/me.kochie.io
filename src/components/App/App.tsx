import * as React from "react";
import Head from 'next/head';

import ReactGA from "react-ga";

import { Sidebar, Main } from "..";

import "./App.css";

ReactGA.initialize("UA-59674915-3");

export default class App extends React.Component {
  componentDidMount() {
    ReactGA.pageview(window.location.pathname + window.location.search);
  }
  
  render() {
    return (
      <div className="app">
        <Head>
          <title>Robert Koch</title>
          
          <meta name="viewport" content="width=device-width, initial-scale=1" />
          <meta charSet="UTF-8" />
          <meta name="keywords" content="Robert Koch, Robert, Koch, Portfolio, Student, Melbourne, Australia" />
          <meta name="description" content="Hi, I'm Robert. This is my site, click through to see some of the content I've worked on. I'm an electrical and software engineer, based in Melbourne, Australia." />
          <meta name="author" content="Robert Koch" />
          <meta name="theme-color" content="#333144" />
          <meta name="google-site-verification" content="I6EWS0ziNrDQdDV1TNtuF1fWq9IFHR767zb6YIFu9wk" />
          
          <link rel="icon" sizes="192x192" href="/static/images/favicon.png" />
          <link href='https://api.mapbox.com/mapbox-gl-js/v0.49.0/mapbox-gl.css' rel='stylesheet' />
          <link href="https://fonts.googleapis.com/css?family=Bowlby+One+SC|Bungee|Source+Sans+Pro|Lato&subset=latin,latin-ext" rel="stylesheet" /> 
        </Head>
        <Sidebar />
        <div className="container">
          <Main />
        </div>
      </div>
    );
  }
}
