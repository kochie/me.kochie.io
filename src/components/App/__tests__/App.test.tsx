import renderer from 'react-test-renderer'
import React from 'react'

import App from '../App'

import { fab } from '@fortawesome/free-brands-svg-icons'
import { library } from '@fortawesome/fontawesome-svg-core'
library.add(fab)

jest.mock('react-ga')

it('app renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
