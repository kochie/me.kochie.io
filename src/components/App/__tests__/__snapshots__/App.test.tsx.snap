// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<div
  className="app"
>
  <div
    className="container"
  >
    <div
      className="sidebar"
    >
      <div
        className="sidebar-header"
      >
        <div
          className="avatar"
        >
          <img
            alt="me"
            src="image-mock.png"
          />
        </div>
        <div
          className="quote"
        >
          I build things, write the code for them, and run from the resulting explosion.
        </div>
        <div
          className="connections"
        >
          <div
            className="grow"
          >
            <a
              aria-label="linkedin"
              className="linkedin"
              href="https://linkedin.com/in/rkkochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-linkedin fa-w-14 fa-2x vertical-middle"
                data-icon="linkedin"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="twitter"
              className="twitter"
              href="https://twitter.com/kochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-twitter fa-w-16 fa-2x vertical-middle"
                data-icon="twitter"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="instagram"
              className="instagram"
              href="https://instagram.com/rkkochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-instagram fa-w-14 fa-2x vertical-middle"
                data-icon="instagram"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="github"
              className="github"
              href="https://github.com/kochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-github fa-w-16 fa-2x vertical-middle"
                data-icon="github"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 496 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="medium-m"
              className="medium"
              href="https://medium.com/@kochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-medium-m fa-w-16 fa-2x vertical-middle"
                data-icon="medium-m"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M71.5 142.3c.6-5.9-1.7-11.8-6.1-15.8L20.3 72.1V64h140.2l108.4 237.7L364.2 64h133.7v8.1l-38.6 37c-3.3 2.5-5 6.7-4.3 10.8v272c-.7 4.1 1 8.3 4.3 10.8l37.7 37v8.1H307.3v-8.1l39.1-37.9c3.8-3.8 3.8-5 3.8-10.8V171.2L241.5 447.1h-14.7L100.4 171.2v184.9c-1.1 7.8 1.5 15.6 7 21.2l50.8 61.6v8.1h-144v-8L65 377.3c5.4-5.6 7.9-13.5 6.5-21.2V142.3z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="gitlab"
              className="gitlab"
              href="https://gitlab.com/kochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-gitlab fa-w-16 fa-2x vertical-middle"
                data-icon="gitlab"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M105.2 24.9c-3.1-8.9-15.7-8.9-18.9 0L29.8 199.7h132c-.1 0-56.6-174.8-56.6-174.8zM.9 287.7c-2.6 8 .3 16.9 7.1 22l247.9 184-226.2-294zm160.8-88l94.3 294 94.3-294zm349.4 88l-28.8-88-226.3 294 247.9-184c6.9-5.1 9.7-14 7.2-22zM425.7 24.9c-3.1-8.9-15.7-8.9-18.9 0l-56.6 174.8h132z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
          <div
            className="grow"
          >
            <a
              aria-label="mastodon"
              className="mastodon"
              href="https://melb.social/@kochie"
            >
              <svg
                aria-hidden="true"
                className="svg-inline--fa fa-mastodon fa-w-14 fa-2x vertical-middle"
                data-icon="mastodon"
                data-prefix="fab"
                focusable="false"
                role="img"
                style={Object {}}
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M433 179.11c0-97.2-63.71-125.7-63.71-125.7-62.52-28.7-228.56-28.4-290.48 0 0 0-63.72 28.5-63.72 125.7 0 115.7-6.6 259.4 105.63 289.1 40.51 10.7 75.32 13 103.33 11.4 50.81-2.8 79.32-18.1 79.32-18.1l-1.7-36.9s-36.31 11.4-77.12 10.1c-40.41-1.4-83-4.4-89.63-54a102.54 102.54 0 0 1-.9-13.9c85.63 20.9 158.65 9.1 178.75 6.7 56.12-6.7 105-41.3 111.23-72.9 9.8-49.8 9-121.5 9-121.5zm-75.12 125.2h-46.63v-114.2c0-49.7-64-51.6-64 6.9v62.5h-46.33V197c0-58.5-64-56.6-64-6.9v114.2H90.19c0-122.1-5.2-147.9 18.41-175 25.9-28.9 79.82-30.8 103.83 6.1l11.6 19.5 11.6-19.5c24.11-37.1 78.12-34.8 103.83-6.1 23.71 27.3 18.4 53 18.4 175z"
                  fill="currentColor"
                  style={Object {}}
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <canvas
        className="canvas"
      />
      <div
        className="bg-image"
      />
    </div>
    <div
      className="main"
    >
      <div
        className="middle"
      >
        <div
          className="content"
        >
          <h1>
            I’m Rob, an engineer from Melbourne, Australia.
          </h1>
          <p>
            I'm a graduate of Monash University where I studied Electrical Engineering, Computer Science, and Mathematics. While studying I was involved with the IT society 
            <a
              href="http://wired.org.au"
            >
              wired
            </a>
            , where I held the title of President in 2017.
          </p>
          <p>
            I currently work as a Cloud Architect for 
            <a
              href="https://aws.amazon.com"
            >
              Amazon Web Services
            </a>
            , where I build solutions for customers in the public sector, think universities, government departments, and non-for-profits. 
          </p>
          <p>
            Before that I worked as a Software Engineer at 
            <a
              href="https://localz.com"
            >
              Localz
            </a>
             who are providing last mile solutions for people around the world.
          </p>
          <p>
            I also helped run the 
            <a
              href="https://unihack.net"
            >
              Unihack
            </a>
             organisation in Melbourne, where we ran hackathons for students around Australia. 
          </p>
          <p>
            I'm an avid runner and enjoy bouldering all around Melbourne and beyond.
          </p>
          <p>
            Apart from that, I love gaming, karate, bike riding, and electronics and try to do at least one of these things every day. I’m always looking for something new and interesting to do, so feel free to browse my portfolio or view my resume if you want to see some of the cool things I’ve done.
          </p>
        </div>
        <div
          className="bmc-button-container"
        >
          <a
            className="bmc-button"
            href="https://www.buymeacoffee.com/kochie"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Buy me a coffee"
              src="https://www.buymeacoffee.com/assets/img/BMC-btn-logo.svg"
            />
            <span
              className="bmc-span"
            >
              Buy me a coffee
            </span>
          </a>
        </div>
        <hr
          className="divider"
        />
        <div
          className="button"
        >
          <a
            download="Robert_Koch.pdf"
            href="/Robert_Koch.pdf"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "margin": "auto",
                "textDecoration": "none",
              }
            }
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-file-user fa-w-12 fa-1x "
              data-icon="file-user"
              data-prefix="fad"
              focusable="false"
              role="img"
              style={
                Object {
                  "--fa-primary-color": "white",
                  "--fa-secondary-color": "#ffdd00",
                  "--fa-secondary-opacity": "1",
                  "stroke": "black",
                  "strokeWidth": "20px",
                }
              }
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                className="fa-group"
                style={Object {}}
              >
                <path
                  className="fa-secondary"
                  d="M384 128H272a16 16 0 0 1-16-16V0H24A23.94 23.94 0 0 0 0 23.88V488a23.94 23.94 0 0 0 23.88 24H360a23.94 23.94 0 0 0 24-23.88V128zm-192 64a64 64 0 1 1-64 64 64 64 0 0 1 64-64zm112 236.8c0 10.61-10 19.2-22.4 19.2H102.4C90 448 80 439.4 80 428.8v-19.2c0-31.81 30.09-57.6 67.1-57.6h5a103.22 103.22 0 0 0 79.7 0h5c37.11 0 67.2 25.79 67.2 57.6z"
                  fill="currentColor"
                  style={Object {}}
                />
                <path
                  className="fa-primary"
                  d="M377 105L279.1 7a24 24 0 0 0-17-7H256v112a16 16 0 0 0 16 16h112v-6.1a23.9 23.9 0 0 0-7-16.9zM192 320a64 64 0 1 0-64-64 64 64 0 0 0 64 64zm44.8 32h-5a103.22 103.22 0 0 1-79.7 0h-5c-37 0-67.1 25.79-67.1 57.6v19.2c0 10.6 10 19.2 22.4 19.2h179.2c12.37 0 22.4-8.59 22.4-19.2v-19.2c0-31.81-30.09-57.6-67.2-57.6z"
                  fill="currentColor"
                  style={Object {}}
                />
              </g>
            </svg>
            <span
              style={
                Object {
                  "color": "white",
                  "fontSize": "25px",
                  "letterSpacing": "-0.6px",
                  "margin": "0 10px",
                }
              }
            >
              Download Resume
            </span>
          </a>
        </div>
        <hr
          className="divider"
        />
        <div
          className="content"
        >
          <div>
            <h2>
              Recent Work
            </h2>
            <div
              className="portfolio"
            >
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://kochie.space"
                  >
                    <img
                      alt="Space"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  Space
                </h3>
                <p>
                  A small webapp that follows the mouse.
                </p>
              </div>
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://3body.now.sh"
                  >
                    <img
                      alt="Particlez"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  Particlez
                </h3>
                <p>
                  A real time n-body particle simulator for the web.
                </p>
              </div>
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://github.com/kochie/guardian-server"
                  >
                    <img
                      alt="Guardian"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  Guardian
                </h3>
                <p>
                  An experiment in transferring cyptographic keys between devices without the user needing to copy keys themselves.
                </p>
              </div>
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://bitbucket.org/kochie/fyp"
                  >
                    <img
                      alt="Optimising Neural Networks for Embedded Systems"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  Optimising Neural Networks for Embedded Systems
                </h3>
                <p>
                  My final year project - focusing on modifying the activation layers of convolutional neural networks to work better on embedded devices.
                </p>
              </div>
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://github.com/kochie/juice"
                  >
                    <img
                      alt="Juice"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  Juice
                </h3>
                <p>
                  A Json-Web-Token analyser.
                </p>
              </div>
              <div
                className="portfolio-item"
              >
                <figure
                  className="portfolio-image-wrapper"
                >
                  <a
                    href="https://github.com/kochie/mcgurrin"
                  >
                    <img
                      alt="McGurrin"
                      className="portfolio-image"
                      src="image-mock.png"
                    />
                  </a>
                </figure>
                <h3>
                  McGurrin
                </h3>
                <p>
                  A tool to help with learning to touch type.
                </p>
              </div>
            </div>
          </div>
        </div>
        <hr
          className="divider"
        />
        <div
          className="content"
        >
          <div>
            <h2>
              Recent Articles
            </h2>
            <div
              className="card"
            >
              <div>
                <a
                  href="https://medium.com/unihack-blog/why-build-an-app-when-you-can-build-a-phone-88b5ce7d0104"
                  rel="noopener"
                >
                  <img
                    alt="Why build an app when you can build a phone?"
                    src="https://cdn-images-1.medium.com/max/800/1*ijymAR61qIf690CHOV3Hcw.jpeg"
                  />
                </a>
              </div>
              <div>
                <h3>
                  Why build an app when you can build a phone?
                </h3>
                <p>
                  Over the last 5 years, UNIHACK has been predominantly focused on software and design. It is not well known that some of UNIHACK’s most creative entire
                  ... 
                  <a
                    href="https://medium.com/unihack-blog/why-build-an-app-when-you-can-build-a-phone-88b5ce7d0104"
                  >
                    continue reading on medium
                  </a>
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <nav
        className="footer"
      >
        <div
          className="name"
        >
          Robert Koch
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-copyright fa-w-16 "
            data-icon="copyright"
            data-prefix="fad"
            focusable="false"
            role="img"
            style={
              Object {
                "padding": "0 5px",
              }
            }
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              className="fa-group"
              style={Object {}}
            >
              <path
                className="fa-secondary"
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm117.13 346.75c-1.59 1.87-39.77 45.73-109.85 45.73-84.69 0-144.48-63.26-144.48-145.56 0-81.31 62-143.4 143.76-143.4 67 0 102 37.31 103.42 38.9a12 12 0 0 1 1.24 14.58l-22.38 34.7a12 12 0 0 1-16.59 3.57 11.79 11.79 0 0 1-1.64-1.27c-.24-.21-26.53-23.88-61.88-23.88-46.12 0-73.92 33.58-73.92 76.08 0 39.61 25.52 79.7 74.28 79.7 38.7 0 65.28-28.34 65.54-28.63a12 12 0 0 1 16.95-.73h.05a12.2 12.2 0 0 1 1.55 1.74l24.55 33.58a12 12 0 0 1-.6 14.85z"
                fill="currentColor"
                style={Object {}}
              />
              <path
                className="fa-primary"
                d="M373.13 354.75c-1.59 1.87-39.77 45.73-109.85 45.73-84.69 0-144.48-63.26-144.48-145.56 0-81.31 62-143.4 143.76-143.4 67 0 102 37.31 103.42 38.9a12 12 0 0 1 1.24 14.58l-22.38 34.7a12 12 0 0 1-16.59 3.57 11.79 11.79 0 0 1-1.64-1.27c-.24-.21-26.53-23.88-61.88-23.88-46.12 0-73.92 33.58-73.92 76.08 0 39.61 25.52 79.7 74.28 79.7 38.7 0 65.28-28.34 65.54-28.63a12 12 0 0 1 16.95-.73h.05a12.2 12.2 0 0 1 1.55 1.74l24.55 33.58a12 12 0 0 1-.6 14.85z"
                fill="currentColor"
                style={Object {}}
              />
            </g>
          </svg>
          2020
        </div>
        <div
          className="guitar"
          id="guitar"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-guitar fa-w-16 "
            data-icon="guitar"
            data-prefix="fad"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              className="fa-group"
              style={Object {}}
            >
              <path
                className="fa-secondary"
                d="M348.88 267.67c7.72 33.39.86 66.87-22 89.75a82.56 82.56 0 0 1-33.47 20.25c-18.54 6-32.67 23.29-34.43 42.15-2.28 23.8-11.48 45.82-28.45 62.7C185 528 103.08 519.94 47.57 464.45s-63.67-137.45-18-183c16.87-17 38.9-26.17 62.59-28.44 18.95-1.77 36.13-15.9 42.15-34.46a82.45 82.45 0 0 1 20.24-33.47c22.9-22.91 56.42-29.81 89.86-22 14.62 3.41 28 10.39 40.15 19.17l-74 74c-.85 0-1.65-.25-2.52-.25a48 48 0 1 0 48 48c0-.84-.21-1.62-.25-2.45l74-74.06c8.74 12.2 15.72 25.51 19.09 40.18z"
                fill="currentColor"
                style={Object {}}
              />
              <path
                className="fa-primary"
                d="M502.63 84.22l-46.32 46.32a35.26 35.26 0 0 1-13.79 8.53l-36.47 12.16-150.33 150.35a47.77 47.77 0 0 0-45.22-45.29L360.79 106 373 69.49a35.26 35.26 0 0 1 8.53-13.79l46.26-46.33a32 32 0 0 1 45.26 0L502.63 39a32 32 0 0 1 0 45.22z"
                fill="currentColor"
                style={Object {}}
              />
            </g>
          </svg>
        </div>
        <div
          className="update"
        >
          <span>
            Updated on 
            <time />
          </span>
        </div>
      </nav>
    </div>
  </div>
</div>
`;
